src/main/java/Steps/ProductPage.java

```diff
@@ -13,60 +15,63 @@
 
     ProductPageLocators productPageLocators = new ProductPageLocators();
     Actions builder = new Actions(BrowserСhoice.getDriver());
-    ExplicitStrategy strategy = new ExplicitStrategy();
 
     public ProductPage() {
         PageFactory.initElements(BrowserСhoice.getDriver(), productPageLocators);
         BrowserСhoice.getDriver().manage().timeouts().implicitlyWait(Duration.ofSeconds(4));
     }
 
-    public String getNameFromFirstProduct(){
-       return productPageLocators.getFirstProductOnPage().getText();
+    public String getNameFromFirstProduct() {
+        return productPageLocators.getFirstProductOnPage().getText();
     }
 
-    public void clickOnFirstProduct(){
+    public void clickOnFirstProduct() {
         productPageLocators.getFirstProductOnPage().click();
     }
 
-    public WebElement returnFirstProduct(){
+    public WebElement returnFirstProduct() {
         return productPageLocators.getFirstProductOnPage();
     }
 
-    public void clickOnBuyButton()  {
-      WebElement waitingElement =  strategy.waitForElement(productPageLocators.getBuyButton());
+    public void clickOnBuyButton() {
+        WebElement waitingElement = ExplicitStrategy.waitForElement(productPageLocators.getBuyButton());
         waitingElement.click();
     }
 
-    public String  orderConfirmText(){
-       return productPageLocators.getOrderConfirm().getText();
+    public String orderConfirmText() {
+        return productPageLocators.getOrderConfirm().getText();
     }
 
     public List<String> findAllProductsAndReturnName() {
-        strategy.waitForElements(productPageLocators.getProducts());
+        ExplicitStrategy.waitForElements(productPageLocators.getProducts());
         return productPageLocators.getProducts().stream().map(WebElement::getText).toList();
     }
 
-    public void addToWishListButton(){
-        productPageLocators.getAddToWishList().click();
+    public void addToWishListButton() {
+        WebElement wishlist = productPageLocators.getAddToWishList();
+        LoginPageLocator loginPage = new LoginPageLocator();
+        ExplicitStrategy.waitForElement(BrowserСhoice.getDriver().findElement(
```

Comment:
I think this wait wont work. It will perform findElement and throw exception before actual wait. You need to use\create method `ExplicitStartegy.waitForElement(By.xpath("...");` which will use `ExpectedConditions.visibilityOfElementLocated()`